test/document.spec.js

```diff
@@ -149,24 +185,51 @@ describe('Document', () => {
   it('should validate that users can update details.', (done) => {
     request.put('/api/documents/Another').set('x-access-token', token).set('Accept', 'application/json').send({
       content: 'This is change'
-    }).expect(200).end((err, res) => {
+    })
+    .expect(200)
+    .end((err, res) => {
       expect(res.body.content).to.equal('This is change');
       done();
     });
   });
 
   it('should validate that users can update details.', (done) => {
+    request.put('/api/documents/Anothers').set('x-access-token', token).set('Accept', 'application/json').send({
+      content: 'To he that does not exist'
+    })
+    .expect(404)
+    .end((err, res) => {
+      expect(res.body.message).to.equal('Document does not exist.');
+      done();
+    });
+  });
+
+  it('should validate that users can update details.', (done) => {
     request.put('/api/documents/Another').set('x-access-token', otherToken).set('Accept', 'application/json').send({
       content: 'This is change'
-    }).expect(200).end((err, res) => {
+    })
+    .expect(200)
+    .end((err, res) => {
       expect(res.body.message).to.equal('Access Denied.');
       done();
     });
   });
 
+  it('should validate that users can only delete documents that exist.', (done) => {
+    request.delete('/api/documents').set('x-access-token', token).set('Accept', 'application/json').send({
+      title: 'tobena'
+    })
+    .expect(400)
```

Comment:
### Critical

If the document does not exist this error code is not the most appropriate because the request was actually well formed.